Python/Product/VSInterpreters/Interpreter/InterpreterRegistryService.cs

```diff
@@ -72,12 +75,21 @@ sealed class InterpreterRegistryService : IInterpreterRegistryService, IDisposab
             }
         }
 
+        public event EventHandler InterpreterDiscoveryCompleted;
+
         private void EnsureFactoryChangesWatched() {
+
             if (!_factoryChangesWatched) {
                 BeginSuppressInterpretersChangedEvent();
                 try {
                     foreach (var provider in GetProviders()) {
                         provider.InterpreterFactoriesChanged += Provider_InterpreterFactoriesChanged;
+
+                        // if the provider is async, listen for the completed event
+                        if (provider is IPythonInterpreterFactoryProviderAsync asyncProvider) {
```

Comment:
I did this in case we add more async providers in the future. I could have just checked if the provider is the conda one (since it's the only async one we have) but I figured this was slightly better.